File: Backend/MyApp.CoreService/Models/Transaction.cs
using MyApp.CoreService.Enums;

namespace MyApp.CoreService.Models;

public class Transaction
{
    public long Id { get; set; }
    public int AccountId { get; set; }
    public TransactionType Type { get; set; }
    public decimal Amount { get; set; }
    public decimal BalanceBefore { get; set; }
    public decimal BalanceAfter { get; set; }
    public int? RelatedAccountId { get; set; }
    public string? Description { get; set; }
    public DateTimeOffset CreatedAt { get; set; }

    public Account Account { get; set; } = null!;
}

File: Mobile/components/user/src/main/java/nekit/corporation/user/domain/model/Transaction.kt
package nekit.corporation.user.domain.model

import java.time.Instant

data class Transaction(
    val id: Long,
    val accountId: Int,
    val type: TransactionTypeDomain,
    val amount: Double,
    val balanceBefore: Double,
    val balanceAfter: Double,
    val relatedAccountId: Int? = null,
    val description: String? = null,
    val createdAt: Instant
)